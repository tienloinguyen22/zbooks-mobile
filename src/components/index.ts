export * from './Body';
export * from './Button';
export * from './Card';
export * from './CardItem';
export * from './Container';
export * from './Content';
export * from './DatePicker';
export * from './ErrorBoundary';
export * from './ErrorText';
export * from './Field';
export * from './Footer';
export * from './Header';
export * from './Icon';
export * from './Image';
export * from './Input';
export * from './Item';
export * from './Label';
export * from './Left';
export * from './List';
export * from './ListItem';
export * from './Loading';
export * from './Picker';
export * from './Right';
export * from './ScrollView';
export * from './Text';
export * from './Touchable';
export * from './Thumbnail';
export * from './Tooltip';
export * from './View';
export * from './WithStore';
export * from './WithScreen';
export * from './Alert';
export * from './Animation';
export * from './AlertContainer';
