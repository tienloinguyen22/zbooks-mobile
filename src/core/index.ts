export * from './i18n';
export * from './themes/colors';
export * from './interfaces/ScreenProps';
export * from './interfaces/LoginType';
export * from './interfaces/User';
export * from './interfaces/Resource';
export * from './interfaces/Theme';
export * from './helpers/show_notification';
export * from './helpers/sleep';
export * from './helpers/record_error';
export * from './helpers/configure_google_sign_in';
export * from './helpers/combine_styles';
export * from './helpers/configure_google_sign_in';
export * from './helpers/handle_error';
export * from './helpers/get_primary_color';
export * from './helpers/format_date';
export * from './screen_names';
