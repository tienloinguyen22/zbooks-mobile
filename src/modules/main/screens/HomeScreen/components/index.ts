export * from './NavigationSample';
export * from './IconSample';
export * from './RematchSample';
export * from './NotificationSample';
export * from './CrashSample';
export * from './PickerSample';
export * from './AnalyticsSample';
export * from './TextStyleSample';
