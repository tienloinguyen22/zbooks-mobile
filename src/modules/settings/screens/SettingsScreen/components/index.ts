export * from './SettingMenuItem';
export * from './LogoutModal';
