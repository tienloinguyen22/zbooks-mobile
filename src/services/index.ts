export * from './app_service';
export * from './auth_service';
export * from './navigation_service';
