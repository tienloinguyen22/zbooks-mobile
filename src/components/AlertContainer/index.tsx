import React from 'react';
import { ViewStyle } from 'react-native';
import Modal from 'react-native-modal';
import { combineStyles, colors, THEME_DARK } from '@app/core';
import { useTranslation } from 'react-i18next';
import { useTheme } from '@app/hooks';
import { Icon } from '../Icon';
import { Button } from '../Button';
import { View } from '../View';
import { styles } from './styles';
import { Text } from '../Text';

export type AlertType = 'SUCCESS' | 'WARNING' | 'ERROR' | 'INFO' | 'DEFAULT';

export interface AlertAction {
  title: string;
  onPress: () => void;
  outline?: boolean;
}

interface AlertProps {
  type: AlertType;
  title?: string;
  message: string;
  actions?: AlertAction[];
  onPressClose: () => void;
  closeable?: boolean;
}

export const AlertContainer = (props: AlertProps): JSX.Element => {
  const { primaryColor, theme } = useTheme();
  const { t } = useTranslation();

  const backgroundAlertStyle = {
    backgroundColor: theme === THEME_DARK ? colors.lightBlack : colors.white,
  };

  const messageTextStyle = {
    color: theme === THEME_DARK ? colors.white : colors.black,
  };

  const closeModal = (): void => {
    props.closeable !== false && props.onPressClose();
  };

  let titleContainerStyle = combineStyles<ViewStyle>(styles.titleDefault, {
    backgroundColor: primaryColor,
  });
  let title = t('dialog.info');
  let iconName = 'information-outline';

  switch (props.type) {
    case 'WARNING':
      titleContainerStyle = combineStyles<ViewStyle>(titleContainerStyle, {
        backgroundColor: colors.orange,
      });
      title = t('dialog.warning');
      iconName = 'alert-circle-outline';
      break;
    case 'ERROR':
      titleContainerStyle = combineStyles<ViewStyle>(titleContainerStyle, {
        backgroundColor: colors.red,
      });
      title = t('dialog.error');
      iconName = 'alert-outline';
      break;
    case 'SUCCESS':
      titleContainerStyle = combineStyles<ViewStyle>(titleContainerStyle, {
        backgroundColor: colors.green,
      });
      title = t('dialog.success');
      iconName = 'check';
      break;
    default:
      break;
  }

  props.title && ({ title } = props);

  return (
    <View style={styles.container}>
      <Modal isVisible={true} onBackButtonPress={closeModal} onBackdropPress={closeModal}>
        <View style={[styles.alertContainer, backgroundAlertStyle]}>
          <View style={titleContainerStyle}>
            <Icon name={iconName} color={colors.white} size={25} />
            <Text style={styles.titleText}>{title}</Text>
            <View style={styles.dialogIcon} />
          </View>
          <Text style={[styles.messageText, messageTextStyle]}>{props.message}</Text>
          {props.actions && (
            <View style={styles.buttonContainer}>
              {props.actions.map((item) => {
                return (
                  <Button outline={item.outline} onPress={item.onPress} key={item.title} style={styles.buttonDialog}>
                    <Text white={!item.outline} primary={item.outline}>
                      {item.title}
                    </Text>
                  </Button>
                );
              })}
            </View>
          )}
          {!props.actions && (
            <View style={styles.reserveContainer}>
              <Button onPress={props.onPressClose} style={styles.buttonDialog}>
                <Text white>{t('dialog.close')}</Text>
              </Button>
            </View>
          )}
        </View>
      </Modal>
    </View>
  );
};
