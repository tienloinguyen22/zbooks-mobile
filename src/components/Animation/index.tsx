import React from 'react';
import { Resource } from '@app/core';
import LottieView from 'lottie-react-native';

interface Props {
  width: number;
  height: number;
  source: Resource;
  autoPlay?: boolean;
  loop?: boolean;
}

export const Animation = (props: Props): JSX.Element => {
  return (
    <LottieView
      source={props.source}
      style={{
        width: props.width,
        height: props.height,
      }}
      autoPlay={props.autoPlay === true}
      loop={props.loop === true}
    />
  );
};
