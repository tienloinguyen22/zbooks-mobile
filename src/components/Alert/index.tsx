import React from 'react';
import RootSiblings from 'react-native-root-siblings';
import { AlertContainer, AlertAction, AlertType } from '../AlertContainer';

interface ShowParams {
  type: AlertType;
  title?: string;
  message: string;
  actions?: AlertAction[];
  onPressClose?: () => void;
  closeable?: boolean;
}

let alert: RootSiblings | undefined;
const hide = (): void => {
  alert && alert.destroy();
};

const show = (props: ShowParams): RootSiblings => {
  alert && hide();
  alert = new RootSiblings(<AlertContainer {...props} onPressClose={props.onPressClose || hide} />);
  return alert;
};

export const Alert = {
  show,
  hide,
};
