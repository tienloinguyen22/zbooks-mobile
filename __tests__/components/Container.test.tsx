import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Container } from '@app/components';
import { navigationService } from '@app/services';
import { THEME_DARK } from '@app/core';
import { mockTheme } from '../test_helpers';

describe('components/Container', () => {
  const headerTitle = 'App header';
  const backButtonText = '';

  beforeEach(() => {
    mockTheme(THEME_DARK);
  });

  it('renders successfully', async () => {
    const { baseElement } = render(<Container />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders successfully with its header', async () => {
    const { baseElement, getByText } = render(<Container showHeader headerTitle={headerTitle} />);

    expect(getByText(headerTitle)).toBeDefined();
    expect(baseElement).toMatchSnapshot();
  });

  it('renders successfully with its header and Back button', async () => {
    const { baseElement, getByText } = render(
      <Container showHeader headerTitle={headerTitle} showBackButton componentId='1' />,
    );

    expect(getByText(headerTitle)).toBeDefined();
    expect(getByText(backButtonText)).toBeDefined();
    expect(baseElement).toMatchSnapshot();
  });

  it('goes back when clicking Back button', async () => {
    navigationService.goBack = jest.fn();
    const { getByText } = render(<Container showHeader headerTitle={headerTitle} showBackButton componentId='1' />);
    const button = getByText(backButtonText);

    fireEvent.press(button);

    expect(navigationService.goBack).toBeCalledTimes(1);
  });
});
