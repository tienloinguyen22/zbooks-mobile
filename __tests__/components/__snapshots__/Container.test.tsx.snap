// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Container renders successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView
      style={
        Array [
          Object {
            "flex": 0,
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    />
    <SafeAreaView
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      />
    </SafeAreaView>
  </View>
</View>
`;

exports[`components/Container renders successfully with its header 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView
      style={
        Array [
          Object {
            "flex": 0,
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    />
    <SafeAreaView
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "height": 50,
            },
            Object {
              "backgroundColor": "black",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 1,
              },
              Object {
                "minWidth": 150,
              },
            ]
          }
        >
          <Text
            bold={true}
            h6={true}
            style={
              Object {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "bold",
                "textTransform": "none",
              }
            }
            white={true}
          >
            App header
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      />
    </SafeAreaView>
  </View>
</View>
`;

exports[`components/Container renders successfully with its header and Back button 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView
      style={
        Array [
          Object {
            "flex": 0,
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    />
    <SafeAreaView
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "height": 50,
            },
            Object {
              "backgroundColor": "black",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 20,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "width": 50,
                }
              }
              transparent={true}
            >
              <Text
                allowFontScaling={false}
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontSize": 25,
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                ÔÖÅ
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 1,
              },
              Object {
                "minWidth": 150,
              },
            ]
          }
        >
          <Text
            bold={true}
            h6={true}
            style={
              Object {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "bold",
                "textTransform": "none",
              }
            }
            white={true}
          >
            App header
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      />
    </SafeAreaView>
  </View>
</View>
`;
